lib/cache-group.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+var Cache = require('./cache');
+
+/*
+ * CacheGroup is used to both speed up, and ensure consistency of hashForDep.
+ *
+ * The CacheGroup contains 3 seperate caches:
+ *
+ *  - MODULE_ENTRY: the cache of PackageCacheEntry objects with information
+ *    about each package found during the hash-for-dep processing.
+ *
+ *    Having the real path means that when resolving depedencies, we can
+ *    take the resolved paths, hash them and check quickly for cache entries,
+ *    speeding creation of the cache.
+ *
+ *    However, that also means we need a second cache to map the
+ *    cacheKey(name, dir) value to the hashed realPath.
+ *
+ *  - PATH: the cache of nameDirKey => realPathKey. nameDirKey is a hash of the
+ *    combination of the name and starting directory passed to hashForDep.
+ *    realPathKey is a hash of the resulting real path to the relevant package.json file.
+ *    See the genRealPathKey() and genNameDirKey() functions.
+ *
+ *  - REAL_FILE_PATH: the cache of a file path to the resolved real path of that file.
```

Comment:
Can we surface the cache key, and cache value of REAL_FILE_PATH cache. Similarly to what you did for PATH above.

Reply:
I need to understand this, will look tomorrow morning.

Reply:
for `PATH` you have `nameDirKey => realPathKey.` which is really nice and descriptive, as to what the cached key/value pair are. 

Could you also surface that here for `REAL_FILE_PATH `